docs: add Codestral x Tabby blog (#2598)
TabbyML/tabby: website/blog/2024-07-09-tabby-codestral/index.mdx

---
authors: [gyxlucy]
tags: [partnership, product]
---

import CodestralVideoUrl from './tabby_codestral.mp4'

> # Introducing the Codestral Integration in Tabby

Reviewer: ```suggestion
# Introducing the Codestral Integration in Tabby

```

We are excited to share the integration of ***Codestral*** model into Tabby! Codestral has demonstrated exceptional performance across various coding tasks, and we are delighted to support our users in both self-hosting and accessing Codestral through its cloud API. This addition aims to enhance your coding experience with precision and efficiency, setting a new standard for AI-enhanced developer experience.

> <video controls width="520" height="340" src={CodestralVideoUrl} type="video/mp4">
> </video>

Reviewer: ```suggestion
<video autoplay muted controls width="520" height="340" src={CodestralVideoUrl} type="video/mp4"/>
```

> ## What's great about Codestral?

Reviewer: ```suggestion
## What's great about Codestral?

```

> Unlike most existing coding LLMs, Codestral stands out with its **shared instruction and completion API endpoint**. This dual capability allows the same model to perform excellently in both code completion and chat tasks, greatly simplifying the developer AI stack and highlighting an important research direction we're extremly excited about. 

Reviewer: ```suggestion
Compared to other coding LLMs, Codestral stands out with its single model that supports both instruction-following and fill-in-the-middle compatibility. This is achieved by fine-tuning the base model on two data sets simultaneously. This dual fine-tuning strategy allows the same model to perform excellently in both code completion and conversational tasks, greatly simplifying the model deployment stack. This highlights an important direction we're extremely excited about from an application perspective, especially in on-prem/byoc use cases.
```

Besides, Codestral is trained on a diverse dataset of **80+ programming languages**, ensuring a high-quality experience for developers working with a wide range of languages.

## Get started with Codestral in Tabby

> ### Self-hosting Codestral

Reviewer: ```suggestion
### Self-hosting Codestral

```

> Codestral-22B is available for both Tabby completion and chat models [link](https://tabby.tabbyml.com/docs/models/). Please follow [Tabby Installation](https://tabby.tabbyml.com/docs/quick-start/installation/) to host Tabby and set `--model Codestral-22B --chat-model Codestral-22B`.

Reviewer: ```suggestion
Codestral-22B is available for Tabby completion [link](https://tabby.tabbyml.com/docs/models/). Please follow [Tabby Installation](https://tabby.tabbyml.com/docs/quick-start/installation/) to host Tabby and set `--model Codestral-22B`.
```

> ### Access via Mistal AI API

Reviewer: ```suggestion
### Access via Mistal AI API

```

> Since [v0.12.0](https://github.com/TabbyML/tabby/releases/tag/v0.12.0), Tabby has supported HTTP API, including Mistral's Codestral API. To use Codestral via Mistral's API, you need to first sign up for an API key from [Mistral AI](https://mistral.ai/). Then in Tabby, set your model configuration as below in `~/.tabby/config.toml` file ([see more](https://tabby.tabbyml.com/docs/administration/model/#mistral--codestral)).

Reviewer: ```suggestion
Tabby also supports connecting to Mistral's API Platform. To use Codestral via Mistral's API, you need to first sign up for an API key from [Mistral AI](https://mistral.ai/). Then in Tabby, set your model configuration as below in `~/.tabby/config.toml` file ([see more](https://tabby.tabbyml.com/docs/administration/model/#mistral--codestral)).
```
